spec/controllers/prompts_controller_spec.rb

```diff
@@ -1,18 +1,136 @@
+    signups = create(:challenge_signup)
+    signups.collection.challenge.signup_open = true
+    signups.collection.challenge.save
+    signups
+  end
 
   before do
     fake_login
   end
 
-  describe 'no_prompt' do
-    it 'should show an error and redirect' do
+  describe "no_challenge" do
+    it "should show an error and redirect" do
+      get :show, collection_id: collection.name
+      it_redirects_to collection_path(collection)
+      expect(flash[:error]).to eq "What challenge did you want to sign up for?"
+    end
+  end
+
+  describe "no_signup" do
+    it "should show an error and redirect" do
+      signups = create(:challenge_signup)
+      post :create, collection_id: signups.collection.name
+      it_redirects_to collection_signups_path(signups.collection) + "/new"
+      expect(flash[:error]).to eq "Please submit a basic sign-up with the required fields first."
+    end
+  end
+
+  describe "signups_closed" do
+    it "should show an error and redirect" do
+      signups = create(:challenge_signup)
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      post :create, collection_id: signups.collection.name
+      it_redirects_to collection_path(signups.collection)
+      expect(flash[:error]).to eq "Signup is currently closed: please contact a moderator for help."
+    end
+  end
+
+  describe "not_signup_owner" do
+    it "should show an error and redirect" do
+      signups = create(:challenge_signup)
+      prompt = signups.collection.prompts.first
+      post :edit, id: prompt.id, collection_id: signups.collection.name
+      it_redirects_to collection_path(signups.collection)
+      expect(flash[:error]).to eq "You can't edit someone else's sign-up!"
+    end
+  end
+
+  describe "new" do
+    context "when prompt_type is offer" do
+      it "should have no errors and redirect" do
+        # Login as the signup owner
+        fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(open_signup.collection).first.pseud_id).user)
+        post :new, collection_id: open_signup.collection.name, prompt_type: "offer"
+        expect(response).to have_http_status(200)
+        expect(flash[:error]).blank?
+      end
+    end
+  end
+
+  describe "create" do
+    it "should have no errors and redirect" do
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(open_signup.collection).first.pseud_id).user)
+      post :create, collection_id: open_signup.collection.name, prompt_type: "offer", prompt: {}
+      it_redirects_to "#{collection_signups_path(open_signup.collection)}/"\
+                      "#{open_signup.collection.prompts.first.challenge_signup_id}/edit"
+      expect(flash[:notice]).blank?
+      expect(flash[:error]).blank?
+    end
+  end
+
+  describe "update" do
+    context "when prompt is valid" do
+      it "should have no errors and redirect" do
```

Comment:
`it "should save the prompt and redirect with a success message"` is probably more representative of the important bits of code than the lack of error.